File: Modules/ServerTypes/PayrollSettings/PayrollEarningsForm.ts
import { DecimalEditor, StringEditor, BooleanEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface PayrollEarningsForm {
    Amount: DecimalEditor;
    EarningCode: StringEditor;
    Description: StringEditor;
    SubjectionToEis: BooleanEditor;
    SubjectionToEpf: BooleanEditor;
    SubjectionToHrdf: BooleanEditor;
    SubjectionToPcb: BooleanEditor;
    SubjectionToSocso: BooleanEditor;
}

export class PayrollEarningsForm extends PrefixedContext {
    static formKey = 'PayrollSettings.PayrollEarnings';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PayrollEarningsForm.init)  {
            PayrollEarningsForm.init = true;

            var w0 = DecimalEditor;
            var w1 = StringEditor;
            var w2 = BooleanEditor;

            initFormType(PayrollEarningsForm, [
                'Amount', w0,
                'EarningCode', w1,
                'Description', w1,
                'SubjectionToEis', w2,
                'SubjectionToEpf', w2,
                'SubjectionToHrdf', w2,
                'SubjectionToPcb', w2,
                'SubjectionToSocso', w2
            ]);
        }
    }
}

File: Modules/HumanResource/PayrollSettings/PayrollEarnings/PayrollEarningsForm.cs
using Serenity.ComponentModel;
using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace HRMSoftware.PayrollSettings.Forms;

[FormScript("PayrollSettings.PayrollEarnings")]
[BasedOnRow(typeof(PayrollEarningsRow), CheckNames = true)]
public class PayrollEarningsForm
{



    [HalfWidth]
    public double Amount { get; set; }
    [HalfWidth]
    public string EarningCode { get; set; }


    public string Description { get; set; }

    [Category("Subjections")]

    [QuarterWidth]
    public int SubjectionToEis { get; set; }

    [QuarterWidth]
    public int SubjectionToEpf { get; set; }

    [QuarterWidth]
    public int SubjectionToHrdf { get; set; }
    [QuarterWidth]
    public int SubjectionToPcb { get; set; }

    [QuarterWidth]
    public int SubjectionToSocso { get; set; }


}